radish34/deploy/src/baseline-administrator-lib/resolvers/contract-resolvers/radish-configpath-resolver.js

```diff
@@ -59,55 +59,52 @@ class RadishConfigpathContractsResolver {
 	 * 
 	 * @param {object} paths - the configuration object. Keys in the object need to be the names of the contracts to be resolved. Values need to be path to the artifact files.
 	 */
-	constructor(paths) {
-		this.paths = DEFAULT_PATHS;
-		if (typeof paths !== 'undefined') {
-			this.paths = require(paths);
-		}
-	}
+  constructor(paths) {
+    this.paths = DEFAULT_PATHS;
+    if (typeof paths !== 'undefined') {
+      this.paths = require(paths);
+    }
+  }
 
 	/**
 	 * Resolves the contract artifacts for a contract with the supplied name
 	 * @param {string} name - the name of the contract to be resolved
 	 * @returns {object} - contains contractName, contract ABI and contract deployment unlinked bytecode
 	 */
-	async resolve(name) {
-		assert(name, 'No name supplied to the resolver resolve method');
-		const contractPath = path.resolve(this.paths[name]);
-		const contractJSON = require(contractPath);
-
-		const contractName = contractJSON.contractName;
-		const abi = contractJSON.compilerOutput.abi;
-		const bytecode = _replaceLibraryReferenceToTruffleStyle(contractJSON.compilerOutput.evm.bytecode.object, contractJSON.compilerOutput.evm.bytecode.linkReferences)
-
-		return {
-			contractName,
-			abi,
-			bytecode
-		}
-	}
+  async resolve(name) {
+    assert(name, 'No name supplied to the resolver resolve method');
+    const contractPath = path.resolve(this.paths[name]);
+    const contractJSON = require(contractPath);
+    const { contractName, abi, bytecode } = contractJSON;
```

Comment:
If the bytecode is comming directly from truffle compile, it might come with its libraries linked already (not sure how they decide what address to use). Please verify, if you have not, that the deployment of contracts with libraries work correctly.

Reply:
As far as I know the library linking is working same as before: the contract bytecode is formatted like this - `0x<contract_source_code_in_hex>__<libraryName>__<library_source_code_in_hex>`

The main difference in deployer is we no longer need to utilize the [_replaceLibraryReferenceToTruffleStyle() function](https://github.com/ethereum-oasis/baseline/blob/master/radish34/deploy/src/baseline-administrator-lib/resolvers/contract-resolvers/radish-configpath-resolver.js#L28) because `truffle` links the libraries within the bytecode in that format by default. Previously, `0x/sol-compiler` output bytecode that linked libraries using the hash of the library instead of the `libraryName` so you were using the `_replaceLibraryReferenceToTruffleStyle` function to convert the hash to the `libraryName`.

I have run the deployer via `make && make start && make test` and the deployer seems to work as well as the current `master` branch deployment. I'm not clear what you mean by "not sure how they decide what [library] address to use". I believe the libraries are part of the newly deployed contract bytecode, not references to previously deployed contracts with their own addresses.

Reply:
Hey Sam, Since it is working I am good.